### ISSUE TITLE
Do not necessarily rely on environment variables

### ISSUE THREAD

**Issue by dmathieu** [neilmiddleton/attr_secure#5]
It'd be cool to not necessarily rely on environment variables.
Having the possibility to store the secret in an other object would be cool.

It would allow to have different keys for different users, each key stored in local storage for example. See [swordfish](https://github.com/github/swordfish) for an other project solving similar problems.

I've been thinking of the following API : 

```
class Foo
  attr_secure :bar, secret_token: lambda {|foo| foo.user.secret_token }
end
```

**Comment by neilmiddleton** [comment]
Agreed.

**Comment by neilmiddleton** [comment]
I have a working version of this here:  https://github.com/neilmiddleton/attr_secure/tree/issue_5

However, the code in attr_secure.rb sucks.

**Comment by dmathieu** [comment]
I've cleaned it up a bit, and added support for lambda secrets. See #7

**Comment by dmathieu** [comment]
Fixed with the merge of #7
